src/generatePagesFromSitemap.ts

```diff
@@ -0,0 +1,119 @@
+import { readFileSync, writeFileSync } from 'node:fs';
+import axios from 'axios';
+import { parseStringPromise } from 'xml2js';
+import yargs from 'yargs';
+import { hideBin } from 'yargs/helpers';
+import { log } from './log';
+import { type PageScreenshotParameter } from './config';
+
+type SitemapParserOptions = {
+  outputPath: string;
+};
+
+type SitemapUrlEntry = {
+  loc: string[];
+  lastmod: string[];
+};
+
+type Sitemap = {
+  urlset: {
+    $: {
+      xmlns: string;
+    };
+    url: SitemapUrlEntry[];
+  };
+};
+
+async function fetchSitemap(url: string): Promise<string> {
+  if (url.startsWith('http')) {
+    const response = await axios.get(url);
+
+    return response.data as string;
+  }
+
+  return readFileSync(url, 'utf8');
+}
+
+async function parseSitemap(sitemapContent: string): Promise<string[]> {
+  const result: Sitemap = (await parseStringPromise(sitemapContent)) as Sitemap;
+
+  if (!result.urlset || !Array.isArray(result.urlset.url)) {
+    throw new Error('Invalid sitemap format');
+  }
+
+  return result.urlset.url
+    .filter(
+      (urlEntry: SitemapUrlEntry) => urlEntry.loc && urlEntry.loc.length > 0,
+    )
+    .map((urlEntry: SitemapUrlEntry) => urlEntry.loc[0]);
+}
+
+async function generatePagesFileFromSitemap(
+  url: string,
+  options: SitemapParserOptions,
+): Promise<void> {
+  try {
+    const sitemapContent = await fetchSitemap(url);
+    const urls = await parseSitemap(sitemapContent);
+
+    const pages: PageScreenshotParameter[] = urls.map((url) => {
+      const page: PageScreenshotParameter = {
+        path: new URL(url).pathname, // Extract the path from the URL
+        name: url.replace(/https?:\/\/|www\./g, '').replace(/\//g, '_'), // Create a unique name
```

Comment:
this will eat up `www` in any place of the URL:

given this list:
```js
[
  'www.example.com', 
  'example.com', 
  'http://example.com', 
  'https://www.example.com', 
  'https://subdomain.www.example.com', 
  'subdomain.www.example.com',
]
```

the result would be:
```js
[
  'example.com', 
  'example.com', 
  'example.com', 
  'example.com', 
  'subdomain.example.com',  // ❌
  'subdomain.example.com',  // ❌
]
```


```suggestion
        name: url.replace(/^https?:\/\/(www\.)?|^www\./g, '').replace(/\//g, '_'), // Create a unique name
```

now you get:
```js
[
  'example.com', 
  'example.com', 
  'example.com', 
  'example.com', 
  'subdomain.www.example.com',  // ✅ 
  'subdomain.www.example.com',  // ✅ 
]